app/models/declaration/DocumentCategory.scala

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package models.declaration
+
+class DocumentCategory(val value: String) {
+  override def equals(obj: Any): Boolean = obj match {
+    case that: DocumentCategory => this.value == that.value
+    case _                      => false
+  }
+
+  override def hashCode(): Int = 31 * (if (value == null) 1 else value.hashCode)
```

Comment:
If you'd make the class abstract, I don't think you need these 2 methods. You are going to extend it in `case object`s anyway and because they are 'case'-ed they should have `equals` and `hashCode` methods correctly implemented. But I am not entirely sure about that.